File: Array/Single Number.go
package main

import "fmt"

func main() {

	arr := []int{1, 2, 2, 3, 1}
	result := singleNumber(arr)
	fmt.Println(result)
}

func singleNumber(A []int) int {
	ans := 0

	for i := 0; i < len(A); i++ {
		ans ^= A[i]
	}

	return ans
}

File: Pythons Solutions/Single Number.py
class Solution:
    def singleNumber(self, arr):
        ans = arr[0]

        for i in range(1, len(arr)):
            ans = ans ^ arr[i]

        return ans

obj = Solution()
arr = [1, 2, 2, 3, 1]
ans = obj.singleNumber(arr)
print(ans)